lib/flatfile.ts

```diff
@@ -248,3 +252,85 @@ export const postFile = async ({
     return null;
   }
 };
+
+export const getRecordsByName = async ({
+  workflow,
+  userId,
+  workbookName,
+  sheetName,
+  spaceType,
+}: {
+  workflow: WorkflowType;
+  userId: string;
+  workbookName: string;
+  sheetName: string;
+  spaceType: SpaceType;
+}) => {
+  const prisma = prismaClient;
+
+  const space = await prisma.space.findUnique({
+    where: {
+      userId_type: {
+        userId,
+        type: spaceType,
+      },
+    },
+  });
+
+  if (!space) {
+    throw new Error(`No space for user ${userId}`);
+  }
+
+  console.log("space", space);
+
+  const { sheetId } = await getWorkbookIdAndSheetIds({
+    workflow,
+    flatfileSpaceId: (space.flatfileData as unknown as FlatfileSpaceData).id,
+    workbookName,
+    sheetName,
+  });
+
+  // console.log("w, s", workbookId, sheetIds);
+
+  const records = await getRecords({ workflow, sheetId });
+
+  return records;
+};
+
+const getWorkbookIdAndSheetIds = async ({
+  workflow,
+  flatfileSpaceId,
+  workbookName,
+  sheetName,
+}: {
+  workflow: WorkflowType;
+  flatfileSpaceId: string;
+  workbookName: string;
+  sheetName: string;
+}): Promise<{ workbookId: string; sheetId: string }> => {
+  const response = await listWorkbooks({
+    workflow,
+    spaceId: flatfileSpaceId,
+  });
+
+  console.log("getWorkbooks response", response);
+
+  const workbookObj = response?.data.find((workbookObj: Workbook) => {
+    return workbookObj.name === workbookName;
+  });
+
+  if (!workbookObj) {
+    throw "Could not find workbook";
+  }
+
+  const sheets = workbookObj.sheets?.find((s) => s.name === sheetName);
+
+  if (!sheets) {
+    throw "Could not find sheets";
```

Comment:
What catches this further up the chain? Should we try/catch here and return a reasonable error + not throw?

Reply:
getRecords needs the sheetId. 

Wrapping the whole function in a try catch